File: frameworks/motia/motia-py/packages/motia/src/motia/enqueue.py
"""Standalone enqueue function for Motia framework."""

from typing import Any

from .iii import get_instance
from .tracing import operation_span


def enqueue(event: dict[str, Any]) -> None:
    """Enqueue an event to a topic."""
    with operation_span("enqueue", **{"motia.step.name": ""}):
        get_instance().trigger({"function_id": "enqueue", "payload": event})

File: frameworks/motia/motia-js/packages/motia/src/new/enqueue.ts
import type { EnqueueData, Enqueuer } from '../types'
import { getInstance } from './iii'

export const enqueue: Enqueuer<EnqueueData> = async (queue: EnqueueData): Promise<void> => {
  await getInstance().trigger({
    function_id: 'enqueue',
    payload: queue,
  })
}
